### PR TITLE
unit_tests

### PR DESCRIPTION
**Pull request by Alex-prokop**
1. Task: https://github.com/rolling-scopes-school/js-fe-course-en/blob/main/tasks/unit-tests/unit-tests.md
2. App screenshot:
Снимок экрана 2023-06-16 в 01 04 46
3. Done 16.06.2023

### CODE REVIEW

File: methods/arrays/chunk.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+function chunk(array, size) {
+  if (array.length <= size) {
+    return [array];
+  }
+  return [array.slice(0, size), ...chunk(array.slice(size), size)];
```
Comment (Xelsoid) [inline methods/arrays/chunk.js]: I suppose .slice method is forbidden 
image

File: methods/arrays/drop.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+function drop(arr, n = 1) {
+  const result = [];
+  if (n < 0) n = 0;
+  return arr.slice(n);
```
Comment (Xelsoid) [inline methods/arrays/drop.js]: the same

File: methods/arrays/drop.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  for (let i = n; i < arr.length; i++) {
+    result.push(arr[i]);
```
Comment (Xelsoid) [inline methods/arrays/drop.js]: the same

File: tests/arrays_test/chunk.test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const { chunk } = require('../../methods/arrays/chunk');
```
Comment (Xelsoid) [inline tests/arrays_test/chunk.test.js:1]: you can import chunk method from lodash and compare the results of execution your method and lodash method

File: tests/arrays_test/chunk.test.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    expect(chunk([true, false, 0, 1, {}, []], 3)).toEqual([
+      [true, false, 0],
+      [1, {}, []],
+    ]);
+  });
+
+  it('should throw an error if the chunk size is negative', () => {
+    expect(() => {
+      chunk([1, 2, 3, 4, 5], -1);
+    }).toThrow();
```
Comment (Xelsoid) [inline tests/arrays_test/chunk.test.js]: to throw what?

File: tests/arrays_test/dropWhile.test.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+const users = [
+  { user: 'barney', active: false },
+  { user: 'fred', active: false },
+  { user: 'pebbles', active: true },
+];
+
+describe('DropWhile Function', () => {
+  it('should drop elements until the predicate function returns false', () => {
+    const result = dropWhile(users, function (o) {
```
Comment (Xelsoid) [inline tests/arrays_test/dropWhile.test.js]: o?

File: tests/objects_test/toPairs.test.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      c: { nested: 'value' },
+    };
+    const result = pickBy(objectWithObjects, (val) => typeof val === 'object');
+    expect(result).toEqual({
+      a: { nested: 'value' },
+      b: { nested: 'value' },
+      c: { nested: 'value' },
+    });
+  });
+});
```
Comment (Xelsoid) [inline tests/objects_test/toPairs.test.js:62]: common comment - I didn't find tests for execution functions with empty arguments
For instance: pick(), pickBy(), omit() and etc

Comment (ArtsiomReutovich) [reply tests/objects_test/toPairs.test.js:62]: optional

**Pull request closed without merging**